lib/automation.go

```diff
@@ -289,13 +289,33 @@ func deploy(scope scope, p *DeployParams) {
 		scope.sendEvent(event.ETAutomationDeployEnd, ref, deployPayload)
 		return
 	}
-	changesSaved = !errors.Is(err, dsfs.ErrNoChanges)
+	if errors.Is(err, dsfs.ErrNoChanges) {
+		ds = p.Dataset
+		changesSaved = false
+		if ds.ID == "" {
+			r := &dsref.Ref{
+				Username: ds.Peername,
+				Name:     ds.Name,
+			}
```

Comment:
I know it says it's deprecated, but use `dsref.ConvertDatasetToVersionInfo(ds).SimpleRef()` instead. We should un-deprecate that function b/c it's totally fine to use, and even necessary in cases like this.

Reply:
fab